autopush/tests/test_router.py

```diff
@@ -508,6 +508,29 @@ def verify_deliver(result):
         d.addCallback(verify_deliver)
         return d
 
+    def test_route_to_busy_node_with_ttl_zero(self):
+        notif = Notification(10, "data", dummy_chid, self.headers, 0)
+        self.agent_mock.request.return_value = response_mock = Mock()
+        response_mock.addCallback.return_value = response_mock
+        type(response_mock).code = PropertyMock(
+            side_effect=MockAssist([202, 200]))
+        self.message_mock.store_message.return_value = True
+        self.message_mock.all_channels.return_value = [dummy_chid]
+        router_data = dict(node_id="http://somewhere", uaid=dummy_uaid)
+        self.router_mock.get_uaid.return_value = router_data
+        self.router.message_id = uuid.uuid4().hex
+
+        d = self.router.route_notification(notif, router_data)
+
+        def verify_deliver(fail):
```

Comment:
silly question: Why is the param called fail if we're looking for a status of 201?